### PR TITLE
Initialize cache in native contracts properly

### LINKED ISSUE #1431: bug: cannot accept new blocks after restoring node from dump
### Problem

I'm running privnet with the current master and patch (66e7d98e0497781fa09b6817ed8de19527eb2509) with the 5th local node. I do the following steps:

1. Stop local node
2. Wait until 4 nodes will process one more block
3. Run local node

Result: local node cannot accept new blocks after restoring from dump:
```
neospcc@neospcc-ThinkPad-T590:~/Documents/GitProjects/nspcc-dev/neo-go$ ./bin/neo-go node -p
2020-09-28T14:05:26.373+0300	INFO	restoring blockchain	{"version": "0.1.0"}
2020-09-28T14:05:26.376+0300	INFO	starting rpc-server	{"endpoint": ":20331"}
2020-09-28T14:05:26.376+0300	INFO	service is running	{"service": "Prometheus", "endpoint": ":2112"}
2020-09-28T14:05:26.376+0300	INFO	service hasn't started since it's disabled	{"service": "Pprof"}
2020-09-28T14:05:26.376+0300	INFO	node started	{"blockHeight": 32, "headerHeight": 33}
2020-09-28T14:05:26.377+0300	INFO	new peer connected	{"addr": "127.0.0.1:20335", "peerCount": 1}
2020-09-28T14:05:26.377+0300	INFO	new peer connected	{"addr": "127.0.0.1:20336", "peerCount": 2}
2020-09-28T14:05:26.377+0300	INFO	new peer connected	{"addr": "127.0.0.1:20334", "peerCount": 3}
2020-09-28T14:05:26.378+0300	INFO	new peer connected	{"addr": "127.0.0.1:20333", "peerCount": 4}
2020-09-28T14:05:26.378+0300	INFO	started protocol	{"addr": "127.0.0.1:20335", "userAgent": "/NEO-GO:0.91.1-pre-237-g66e7d98e/", "startHeight": 66, "id": 4217096708}
2020-09-28T14:05:26.379+0300	INFO	started protocol	{"addr": "127.0.0.1:20334", "userAgent": "/NEO-GO:0.91.1-pre-237-g66e7d98e/", "startHeight": 66, "id": 823238801}
2020-09-28T14:05:26.379+0300	INFO	started protocol	{"addr": "127.0.0.1:20336", "userAgent": "/NEO-GO:0.91.1-pre-237-g66e7d98e/", "startHeight": 66, "id": 4205287763}
2020-09-28T14:05:26.380+0300	INFO	started protocol	{"addr": "127.0.0.1:20333", "userAgent": "/NEO-GO:0.91.1-pre-237-g66e7d98e/", "startHeight": 66, "id": 2691420827}
2020-09-28T14:05:26.384+0300	WARN	blockQueue: failed adding block into the blockchain	{"error": "postPersist failed: VM has failed: error encountered at instruction 5 (SYSCALL): runtime error: index out of range [1] with length 0", "blockHeight": 32, "nextIndex": 33}
2020-09-28T14:05:26.386+0300	WARN	blockQueue: failed adding block into the blockchain	{"error": "postPersist failed: VM has failed: error encountered at instruction 5 (SYSCALL): runtime error: index out of range [1] with length 0", "blockHeight": 32, "nextIndex": 33}

```

### PR DESCRIPTION
**Pull request by fyrchik**
Introduced in bd1790bd , 83e94d3b .
Fix #1431.
Testing done:
1. Create dump.
2. Run node from restored dump.

### CODE REVIEW

**Review by AnnaShaleva — APPROVED** [review]

**Comment by AnnaShaleva** [comment]
It also solves the bug described in https://github.com/nspcc-dev/neo-go/pull/1430.

File: pkg/core/native/native_neo.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ type NEO struct {
 	gasPerBlock        atomic.Value
 	gasPerBlockChanged atomic.Value
 
+	initDone       bool
```
Comment (roman-khimov) [inline pkg/core/native/native_neo.go:35]: Yet another flag? Can we arrange to live without it?

Comment (fyrchik) [reply pkg/core/native/native_neo.go:35]: We can reuse e.g. `gasPerBlockChanged` and check if it is `nil` in `OnPersist`, but it seems a bit clunky.

Comment (fyrchik) [reply pkg/core/native/native_neo.go]: Fixed.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**